launchpad/reward_calculation.gno

```diff
@@ -94,18 +117,14 @@ func (self *RewardState) CalculateReward(depositId string) uint64 {
 	info := self.Info(depositId)
 	stakerPrice := u256.Zero().Sub(self.PriceAccumulation, info.PriceDebt)
 	reward := stakerPrice.Mul(stakerPrice, u256.NewUint(info.Amount))
-	reward = reward.Rsh(reward, 128)
-	return reward.Uint64()
+	reward = reward.Rsh(reward,128)
+	return reward.Uint64() - info.Claimed
 }
 
-func (self *RewardState) PriceAccumulationUint64() uint64 {
-	return u256.Zero().Rsh(self.PriceAccumulation, 128).Uint64()
-}
-
-
 // amount MUST be less than or equal to the amount of xGNS staked
 // This function does not check it
```

Comment:
if this function doesn't check, then which function verify it??

Reply:
This code is exactly the same as the implementing reward calculations in `gov/staker`. In that contract, the `deductReward` function is validating its functionality with `RemoveStake` method.